kotlin/android/app/src/main/java/dev/firezone/android/core/data/Repository.kt

```diff
@@ -49,13 +53,22 @@ fun ResourceState.toggle(): ResourceState {
     }
 }
 
+/// Wrapper class used because `MutableStateFlow` will not
+/// notify subscribers if you submit the same object that's already in it.
+class Favorites(val inner: HashSet<String>);
```

Comment:
SO indicates we could also try `MutableSharedFlow` here, but the wrapper was easy to set up and I already had `MutableStateFlow` in place.